Sources/ProjectSpec/SwiftPackage.swift

```diff
@@ -7,24 +7,38 @@ public struct SwiftPackage: Equatable {
 
     public typealias VersionRequirement = XCRemoteSwiftPackageReference.VersionRequirement
 
-    public let url: String
-    public let versionRequirement: VersionRequirement
+    public enum Kind: Equatable {
+        case remote(url: String, versionRequirement: VersionRequirement)
+        case local(path: String)
+        
+        public var isLocal: Bool {
+            if case .local = self {
+                return true
+            }
+            return false
+        }
+    }
+
+    public let kind: SwiftPackage.Kind
 
-    public init(url: String, versionRequirement: VersionRequirement) {
-        self.url = url
-        self.versionRequirement = versionRequirement
+    public init(kind: SwiftPackage.Kind) {
+        self.kind = kind
     }
 }
 
 extension SwiftPackage: JSONObjectConvertible {
 
     public init(jsonDictionary: JSONDictionary) throws {
-        url = try jsonDictionary.json(atKeyPath: "url")
-        versionRequirement = try VersionRequirement(jsonDictionary: jsonDictionary)
-        try validateVersion()
+        if let url: String = jsonDictionary.json(atKeyPath: "url") {
```

Comment:
Let's reverse this so if `path` is present, parse as local, otherwise remote. This let's people simply add a path to a remote definition to temporarily make it local, without having to remove url and version